### PR TITLE
Convert to Rust worker

### PR DESCRIPTION
**Pull request by tolbrino**
- Discarded old endpoints which were not used anymore
- Added new endpoint `gnosis` as a future replacement for `xdai_mainnet`
- Added auth token support which is required for some endpoints

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,14 +1,32 @@
 {
   description = "provider proxy";
 
-  inputs.flake-utils.url = "github:numtide/flake-utils";
   inputs.nixpkgs.url = github:NixOS/nixpkgs/nixpkgs-unstable;
```
Comment (Teebor-Choka) [inline flake.nix:4]: Should we not prefer stable?

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+use crate::providers::get_provider;
+use hyper;
+use std::str::FromStr;
+use std::str::Utf8Error;
+
+pub use console_error_panic_hook::set_once as set_panic_hook;
+
+use worker::*;
+
+fn map_utf8_error(error: Utf8Error) -> worker::Error {
```
Comment (Teebor-Choka) [inline src/lib.rs:38]: These should be idiomatically implemented as `impl From<A> for B`

File: src/providers.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+fn provider(name: &str, endpoints: Vec<ProviderEndpoint>) -> Provider {
+    Provider {
+        name: name.to_string(),
+        endpoints,
+    }
+}
+
+fn endpoint(url: &str, auth_token: Option<&str>) -> ProviderEndpoint {
```
Comment (Teebor-Choka) [inline src/providers.rs:20]: To simplify our life though, AAA could be delegated to a reverse proxy.

Comment (tolbrino) [reply src/providers.rs:20]: It will be eventually. Until then this was a simple change.

File: flake.nix (hunk 1 above)
Comment (tolbrino) [reply flake.nix:4]: unstable contains the recent wasm-pack etc

File: src/lib.rs (hunk 2 above)
Comment (tolbrino) [reply src/lib.rs:38]: Cannot easily do that because of https://doc.rust-lang.org/error_codes/E0117.html

**Review by ausias-armesto — APPROVED** [review]

**Pull request merged**